### PR TITLE
Add deletion part

### PR DESCRIPTION
**Pull request by hananeikhelef**
**Done**
- Add deletion part to the comment section, now we can delete a comment 

**TODO**
- Add authentication part

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -14,41 +14,48 @@
...
-
-    for(const x in stats){
-        statsListElement.appendChild(
-        createListElement(stats[x]));
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const messageListElement = document.getElementById('message-container');    
+    messageListElement.innerHTML="";
+    
+    for(const x in messages){
+        if(messages[x].content !== undefined){
```
Comment (mctavish) [inline portfolio/src/main/webapp/script.js:31]: Is this a risk?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -14,41 +14,48 @@
...
-    statsListElement.innerHTML = '';
-
-    for(const x in stats){
-        statsListElement.appendChild(
-        createListElement(stats[x]));
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const messageListElement = document.getElementById('message-container');    
+    messageListElement.innerHTML="";
+    
+    for(const x in messages){
```
Comment (mctavish) [inline portfolio/src/main/webapp/script.js:30]: Consider using for...of
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -14,41 +14,48 @@
...
+
+  messageElement.appendChild(deleteButtonElement);
+  return messageElement;
+}
+
+/** Tells the server to delete the comment. */
+function deleteMessage(message) {
+  const params = new URLSearchParams();
+  params.append('id', message);
+  fetch('/delete-data', {method: 'POST', body: params});
```
Comment (mctavish) [inline portfolio/src/main/webapp/script.js:60]: No action needed, but consider what should happen if this call fails.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (hananeikhelef) [reply portfolio/src/main/webapp/script.js:31]: Yes, while i was debugging i notices that even when there's en empty message it still considers it.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (hananeikhelef) [reply portfolio/src/main/webapp/script.js:30]: Will do, thank you !

**Pull request merged**